scripts/base/protocols/ldap/main.zeek

```diff
@@ -0,0 +1,513 @@
+  # This is the format of ldap.log (ldap operations minus search-related)
+  # Each line represents a unique connection+message_id (requests/responses)
+  type Message: record {
+
+    # Timestamp for when the event happened.
+    ts: time &log;
+
+    # Unique ID for the connection.
+    uid: string &log;
+
+    # The connection's 4-tuple of endpoint addresses/ports.
+    id: conn_id &log;
+
+    # transport protocol
+    proto: string &log &optional;
+
+    # Message ID
+    message_id: int &log &optional;
+
+    # LDAP version
+    version: int &log &optional;
+
+    # normalized operations (e.g., bind_request and bind_response to "bind")
+    opcode: set[string] &log &optional;
+
+    # Result code(s)
+    result: set[string] &log &optional;
+
+    # result diagnostic message(s)
+    diagnostic_message: vector of string &log &optional;
+
+    # object(s)
+    object: vector of string &log &optional;
+
+    # argument(s)
+    argument: vector of string &log &optional;
+  };
+
+  #############################################################################
+  # This is the format of ldap_search.log (search-related messages only)
+  # Each line represents a unique connection+message_id (requests/responses)
+  type Search: record {
+
+    # Timestamp for when the event happened.
+    ts: time &log;
+
+    # Unique ID for the connection.
+    uid: string &log;
+
+    # The connection's 4-tuple of endpoint addresses/ports.
+    id: conn_id &log;
+
+    # transport protocol
+    proto: string &log &optional;
+
+    # Message ID
+    message_id: int &log &optional;
+
+    # sets of search scope and deref alias
+    scope: set[string] &log &optional;
+    deref: set[string] &log &optional;
+
+    # base search objects
+    base_object: vector of string &log &optional;
+
+    # number of results returned
+    result_count: count &log &optional;
+
+    # Result code (s)
+    result: set[string] &log &optional;
+
+    # result diagnostic message(s)
+    diagnostic_message: vector of string &log &optional;
+
+    #  a string representation of the search filter used in the query
+    filter: string &log &optional;
+
+    # a list of attributes that were returned in the search
+    attributes: vector of string &log &optional;
+  };
+
+  # Event that can be handled to access the ldap record as it is sent on
+  # to the logging framework.
+  global log_ldap: event(rec: LDAP::Message);
+  global log_ldap_search: event(rec: LDAP::Search);
+
+  # Event called for each LDAP message (either direction)
+  global LDAP::message: event(c: connection,
+                              message_id: int,
+                              opcode: LDAP::ProtocolOpcode,
+                              result: LDAP::ResultCode,
+                              matched_dn: string,
+                              diagnostic_message: string,
+                              object: string,
+                              argument: string);
+
+  const PROTOCOL_OPCODES = {
```

Comment:
Conventionally these would go into a `ldap/const.zeek` file and `main.zeek` would be a bit shorter.